ibazel/fsevents_watcher/fsevents_watcher.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build darwin
+
+package fsevents_watcher
+
+import (
+	"github.com/fsnotify/fsevents"
+
+	"github.com/bazelbuild/bazel-watcher/ibazel/fs_watcher"
+)
+
+type realFSEventsWatcher struct {
+	es *fsevents.EventStream
+	evs chan fs_watcher.Event
+}
+
+var _ fs_watcher.Watcher = &realFSEventsWatcher{}
```

Comment:
When you move this inside `github.com/bazelbuild/bazel-watcher/ibazel/fswatcher/fseventswatcher` you'll need to remove this line due to circular imports, but that's okay, the interface will be verified by the other package.